### ISSUE TITLE
Add `allowPairs` option to `vue/max-attributes-per-line` (for `multiline.max == 1`)

### ISSUE THREAD

**Issue by dmke** [vuejs/eslint-plugin-vue#2122]
**What rule do you want to change?**

- `vue/max-attributes-per-line`

**Does this change cause the rule to produce more or fewer warnings?**

- Likely more, but only when active.

**How will the change be implemented? (New option, new default behavior, etc.)?**

- This should be a new option to `multiline`.

**Please provide some example code that this change will affect:**

```ts
interface MaxAttributesPerLineOptions {
  singleline: {
    max: number
  } | number

  multiline: {
    max: number,
    allowPairs?: [string, string][] // NEW: optional list of 2-element arrays
  } | number
}
```

```js
// .eslintrc.js
{
  rules: {
    "vue/max-attributes-per-line": ["error", {
      singleline: 2,
      multiline: {
        max: 1,             // restrict to one attribute per line
        allowPairs: [
          ["v-for", "key"], // allow `v-for="i in list" :key="i"` in a single line
        ],
      },
    }],
  }
}
```

```vue
<!-- wrong: singleline.max violated -->
<ThingItem v-for="thing of things" :key="thing.id" class="thing" />

<!-- wrong: multiline.pairings[0] violated -->
<ThingItem
  v-for="thing of things"
  :key="thing.id"
  class="thing"
/>

<!-- correct -->
<ThingItem
  v-for="thing of things" :key="thing.id"
  class="thing"
/>
```

**What does the rule currently do for this code?**

We currently have the following configuration, in order to allow `v-for` and `key` on the same line:

```js
{
  rules: {
    "vue/max-attributes-per-line": ["error", {
      singleline: 2,
      multiline:  2,
    }],
  }
}
```

However that will also allow any other two attributes on the same line, which needs manual intervention:

```vue
<ThingItem
  v-for="thing of things" :class="{ active: thing.active }"
  :key="thing.id" class="thing"
/>
```

**What will the rule do after it's changed?**

- When `multiline.max >= 2`, nothing changes.
- When `multiline.max == 1` and `multiline.allowPairs == null | undefined | []`, nothing changes.
- Otherwise (`multiline.max == 1` and `multiline.allowPairs` is a non-empty array), the corrections shown above are applied

**Additional context**

- I don't really care about the option name; `allowPairs`, `allowPairings`, `exceptions` are all fine by me.
- I don't know if it is necessary to make the list of attributes configurable.
  - The only other use case I can find in my code base is something like `class="item" :class={ active }` - and I'd be OK with this split to multiple lines.

**Comment by FloEdelmann** [comment]
This is really a special case. I'm not sure this is worth the effort.

What exactly is the benefit of having `v-for` and `key` in the same line instead of having them in separate, consecutive lines?

**Comment by FloEdelmann** [comment]
> if things belong together, they should also be removed together

Yes, but sometimes other attributes also belong together. It's hard to draw a line there. So rather disabling or relaxing the `multiline.max` option seems better in your case.

> "unnecessary amount" of scrolling

Only combining `v-for` and `key` lines won't help much then. Again, this seems to be an argument for relaxing your `multiline.max` configuration generally.

**Comment by dmke** [comment]
> > if things belong together, they should also be removed together
> 
> Yes, but sometimes other attributes also belong together. It's hard to draw a line there. So rather disabling or relaxing the `multiline.max` option seems better in your case.

Yeah, I'm leaning that way as well. Is there a rule that sort props, and does it group `v-for` and `:key`?

> > "unnecessary amount" of scrolling
> 
> Only combining `v-for` and `key` lines won't help much then. Again, this seems to be an argument for relaxing your `multiline.max` configuration generally.

Sure. It was a straw-man argument anyway.

I'll close this issue for now. Maybe I can find a stronger argument in the future :)

**Comment by FloEdelmann** [comment]
> Is there a rule that sort props, and does it group `v-for` and `:key`?

Yes and yes (it is configurable): https://eslint.vuejs.org/rules/attributes-order.html

**Comment by dmke** [comment]
Ah! I've been searching for "prop sorting" and only found `vue/sort-keys` (which sorts the keys in Options API component definitions).

Thank you for the pointer!

**Comment by FloEdelmann** [comment]
Would you fancy opening a PR to improve the docs to include the keyword?
